src/controller/python/chip/bdx/bdx-transfer.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <controller/python/chip/bdx/bdx-transfer.h>
+
+#include <protocols/bdx/BdxTransferSession.h>
+
+namespace chip {
+namespace bdx {
+namespace {
+
+constexpr uint32_t kMaxBdxBlockSize               = 1024;
+constexpr System::Clock::Timeout kBdxPollInterval = System::Clock::Milliseconds32(50);
+constexpr System::Clock::Timeout kBdxTimeout      = System::Clock::Seconds16(5 * 60);
+
+} // namespace
+
+void BdxTransfer::SetDelegate(BdxTransfer::Delegate * delegate)
+{
+    mDelegate = delegate;
+}
+
+BdxTransfer::~BdxTransfer()
+{
+    if (mData != nullptr)
+    {
+        delete[] mData;
+        mData = nullptr;
+    }
+}
+
+CHIP_ERROR BdxTransfer::AcceptSend()
+{
+    VerifyOrReturnError(mAwaitingAccept, CHIP_ERROR_INCORRECT_STATE);
+    mAwaitingAccept = false;
+
+    TransferSession::TransferAcceptData acceptData;
+    acceptData.ControlMode  = TransferControlFlags::kSenderDrive;
+    acceptData.MaxBlockSize = mTransfer.GetTransferBlockSize();
+    acceptData.StartOffset  = mTransfer.GetStartOffset();
+    acceptData.Length       = mTransfer.GetTransferLength();
+    return mTransfer.AcceptTransfer(acceptData);
+}
+
+CHIP_ERROR BdxTransfer::AcceptReceive(const ByteSpan & data_to_send)
+{
+    VerifyOrReturnError(mAwaitingAccept, CHIP_ERROR_INCORRECT_STATE);
+    mAwaitingAccept = false;
+    mData           = new uint8_t[data_to_send.size()];
+    if (mData == nullptr)
+    {
+        return CHIP_ERROR_NO_MEMORY;
+    }
+    memcpy(mData, data_to_send.data(), data_to_send.size());
```

Comment:
Is there a copy function in span.h that would work here? We're trying to get folks to stay away from raw memcpys with spans because we've had a couple of solid bugs due to off by ones.